compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -536,6 +536,8 @@ impl<'a> Parser<'a> {
         } else if inedible.contains(&self.token.kind) {
             // leave it in the input
             Ok(false)
+        } else if self.token.kind == token::Eof {
+            self.expected_one_of_not_found(edible, inedible)
         } else if self.last_unexpected_token_span == Some(self.token.span) {
```

Comment:
```suggestion
        } else if self.token.kind != token::Eof && self.last_unexpected_token_span == Some(self.token.span) {
```

I think this is better, we avoid duplicating the content of the `else` block, which is the behaviour we want.